src/utils/test/packet_collectors.rs

```diff
@@ -33,3 +34,39 @@ impl<T: Debug> Future for ExhaustiveDrain<T> {
         }
     }
 }
+
+/// Exhaustive Collector works like Exhaustive Drain in that it continously polls for packets until it
+/// receives a None, but it also will write that value out to the provided channel, so that the packet
+/// may be compared in a test.
+pub struct ExhaustiveCollector<T: Debug> {
+    id: usize,
+    stream: ElementStream<T>,
+    packet_dump: crossbeam_channel::Sender<T>
+}
+
+impl<T: Debug> ExhaustiveCollector<T> {
+    pub fn new(id: usize, stream: ElementStream<T>, packet_dump: crossbeam_channel::Sender<T>) -> Self {
```

Comment:
In other places we've been typing by just `Sender<T>`. Could we import the whole `crossbeam::crossbeam_channel::Sender` please?

Reply:
Will clean this up.  I'm going to go with calls to `bounded` will be `crossbeam_channel::bounded`,  calls to Sender and Receiver will be `Sender` and `Receiver`.   Similar pattern for unbounded.  I think that balances readability with succinctness.

Reply:
Cleaned up as requested. Future PRs will also follow this convention.